compiler/src/verify.rs

```diff
@@ -16,7 +18,13 @@ pub fn compile_asm_string_temp<T: FieldElement>(
     contents: &str,
     inputs: Vec<T>,
 ) -> (String, Temp) {
-    let pil = pilgen::compile(Some(file_name), contents).unwrap();
+    let parsed = parser::parse_asm(Some(file_name), contents).unwrap_or_else(|err| {
+        eprintln!("Error parsing .asm file:");
+        err.output_to_stderr();
+        panic!();
+    });
+    let analysed = analyse(parsed).unwrap();
+    let pil = convert(analysed);
```

Comment:
Maybe `convert` could do the parsed and analysed parts inside itself? Or did you keep it separated on purpose?